expression/builtin_string.go

```diff
@@ -453,38 +453,45 @@ func (b *builtinLowerSig) eval(row []types.Datum) (d types.Datum, err error) {
 	}
 }
 
+func oneArgsInferType(args []Expression) *types.FieldType {
+	argTp := args[0].GetType()
+	tp := types.MergeFieldType(mysql.TypeVarString, argTp.Tp)
+	retType := types.NewFieldType(tp)
+	retType.Charset, retType.Collate = charset.CharsetUTF8, charset.CollationUTF8
+	if types.IsBinaryStr(argTp) {
+		retType.Charset, retType.Collate = charset.CharsetBin, charset.CollationBin
+		retType.Flag |= mysql.BinaryFlag
+	}
+	return retType
```

Comment:
please also set `retType`'s `Flen` before return or during the evaluation of `builtinReverseSig`